pkg/k8s/watchers/endpointsynchronizer.go

```diff
@@ -130,34 +130,37 @@ func (epSync *EndpointSynchronizer) RunK8sCiliumEndpointSync(e *endpoint.Endpoin
 						return nil
 					}
 
-					scopedLog.Debug("Deleting CEP during an initialization")
-					err := ciliumClient.CiliumEndpoints(namespace).Delete(ctx, podName, meta_v1.DeleteOptions{})
```

Comment:
I think this was done a long time ago when we changed the format of the CIliumEndpoint at the very beginning for this feature and this allowed to up and downgrade.